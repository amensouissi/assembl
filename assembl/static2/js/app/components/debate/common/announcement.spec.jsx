import React from 'react';
import renderer from 'react-test-renderer'; // eslint-disable-line

import Announcement, { createDoughnutElements } from '../../../../../js/app/components/debate/common/announcement';

describe('Announcement component', () => {
  const announcementContent = {
    body: '<p>Bonjour Ã  tous</p>'
  };
  const fakePost1 = {
    __typename: 'PostEdge',
    node: {
      __typename: 'Post',
      id: '9706',
      parentId: 'null',
      creationDate: '2018-02-21T18:07:49.630164+00:00',
      publicationState: 'PUBLISHED',
      originalLocale: 'fr',
      sentimentCounts: { like: 2, disagree: 1, dontUnderstand: 0, moreInfo: 1, __typename: 'SentimentCounts' }
    }
  };
  const fakePost2 = {
    __typename: 'PostEdge',
    node: {
      __typename: 'Post',
      id: '9706',
      parentId: '9706',
      creationDate: '2018-02-21T18:08:14.590132+00:00',
      publicationState: 'PUBLISHED',
      originalLocale: 'fr',
      sentimentCounts: { like: 1, disagree: 1, dontUnderstand: 9, moreInfo: 0, __typename: 'SentimentCounts' }
    }
  };
  describe('<Announcement/> on a thread', () => {
    it('should render an announcement with the content associated to a thread idea', () => {
      const threadIdea = {
        id: '1234',
        messageColumns: [],
        messageViewOverride: null,
        numContributors: 2,
        numPosts: 2,
        posts: { edges: [fakePost1, fakePost2] },
        __typename: 'Idea'
      };
      const announcementProps = {
        announcementContent: announcementContent,
        idea: threadIdea,
        isMultiColumns: false
      };
      const component = renderer.create(<Announcement {...announcementProps} />);
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
  describe('<Announcement /> on a multiColumns', () => {
    it('should render an announcement with the content associated to a multiColumns idea', () => {
      const positiveColumn = {
        color: 'green',
        columnSynthesis: {},
        index: 0,
        messageClassifier: 'positive',
        name: 'Positive',
        numPosts: 1,
        title: 'Positive',
        __typename: 'IdeaMessageColumn'
      };
      const negativeColumn = {
        color: 'red',
        columnSynthesis: {},
        index: 1,
        messageClassifier: 'negative',
        name: 'Negative',
        numPosts: 1,
        title: 'Negative',
        __typename: 'IdeaMessageColumn'
      };
      const messageColumns = [positiveColumn, negativeColumn];
      const multiColumnsIdea = {
        id: '12345',
        messageColumns: messageColumns,
        messageViewOverride: 'messageColumns',
        numContributors: 2,
        numPosts: 2,
        posts: {
          edges: [{ ...fakePost1, messageClassifier: 'positive' }, { ...fakePost2, messageClassifier: 'negative' }]
        }
      };
      const announcementProps = {
        announcementContent: announcementContent,
        idea: multiColumnsIdea,
        isMultiColumns: true
      };
      const component = renderer.create(<Announcement {...announcementProps} />);
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
  describe('createDoughnutElements function', () => {
    it('should return an array of elements for a <StatisticDoughnut /> component', () => {
      const fakeSentimentCounts = {
        dontUnderstand: {
          type: 'DONT_UNDERSTAND',
          camelType: 'dontUnderstand',
          color: 'yellow',
          tooltip: {},
          count: 2,
          svgComponent: {
            defaultProps: {
              backgroundColor: '#ffffff',
              color: 'purple'
            }
          }
        },
        disagree: {
          type: 'DISAGREE',
          camelType: 'disagree',
          color: 'red',
          tooltip: {},
          count: 0,
          svgComponent: {
            defaultProps: {
              backgroundColor: '#ffffff',
              color: 'purple'
            }
          }
        },
        like: {
          type: 'LIKE',
          camelType: 'like',
          color: 'green',
          tooltip: {},
          count: 4,
          svgComponent: {
            defaultProps: {
              backgroundColor: '#ffffff',
              color: 'purple'
            }
          }
        },
        moreInfo: {
          type: 'MORE_INFO',
          camelType: 'moreinfo',
          color: 'purple',
          tooltip: {},
          count: 3,
          svgComponent: {
            defaultProps: {
              backgroundColor: '#ffffff',
              color: 'purple'
            }
          }
        }
      };
      const result = createDoughnutElements(fakeSentimentCounts);
      expect(result).toMatchSnapshot();
    });
  });
});