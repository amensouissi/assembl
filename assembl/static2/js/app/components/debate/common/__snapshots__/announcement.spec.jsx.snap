// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Announcement component <Announcement /> on a multiColumns should render an announcement with the content associated to a multiColumns idea 1`] = `
<div
  className="announcement"
>
  <div
    className="announcement-title"
  >
    <div
      className="title-hyphen"
    >
       
    </div>
    <h3
      className="announcement-title-text dark-title-1"
    >
      <span
        className={undefined}
        style={undefined}
      >
        Announcement
      </span>
    </h3>
  </div>
  <div
    className="announcement-media col-md-push-4 col-md-8 col-xs-12"
  >
    <section
      className="media-section background-light"
    >
      <div
        className="max-container"
      >
        <div
          className="container-fluid"
        >
          <div
            className="col-md-4 col-sm-12"
          >
            <div
              className="media-description"
            >
              <div
                className="media-description-icon"
              >
                <span
                  className="assembl-icon-pepite color2"
                >
                   
                </span>
              </div>
              <div
                className="description-txt"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": undefined,
                  }
                }
              />
              <div
                className="box-hyphen left"
              >
                 
              </div>
            </div>
          </div>
          <div
            className="col-md-8 col-sm-12"
          >
            <div
              className="media-right"
            >
              <div
                className="media-description-layer media-description-top"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "<p>Bonjour à tous</p>",
                  }
                }
              />
              <div
                className="media-description"
              >
                <div
                  className="media-description-icon"
                >
                  <span
                    className="assembl-icon-pepite color2"
                  >
                     
                  </span>
                </div>
                <div
                  className="description-txt"
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": undefined,
                    }
                  }
                />
                <div
                  className="box-hyphen left"
                >
                   
                </div>
              </div>
              <div
                className="media-description-layer media-description-bottom"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": null,
                  }
                }
              />
            </div>
          </div>
        </div>
      </div>
    </section>
  </div>
  <div
    className="col-md-pull-8 col-md-4 col-xs-12"
  >
    <div
      className="announcement-statistics"
    >
      <div
        className="announcement-doughnut"
      >
        <div
          className="statistics-doughnut"
        >
          <div
            className="statistics"
          >
            <div
              className="doughnut-container superpose"
            >
              <svg
                className="doughnut"
                viewBox="0 0 140 140"
              >
                <path
                  className="circle"
                  d="M 70 120 A 50 50 0 0 0 70 20"
                  fill="transparent"
                  stroke="green"
                />
                <path
                  className="circle"
                  d="M 69.99999999999999 20 A 50 50 0 0 0 70 120"
                  fill="transparent"
                  stroke="red"
                />
              </svg>
            </div>
            <div
              className="doughnut-label superpose"
            >
              <div
                className="doughnut-label-count"
              >
                2
              </div>
              
              <div
                className="doughnut-label-text"
              >
                <span
                  className={undefined}
                  style={undefined}
                >
                  Reactions
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="announcement-numbers-multicol"
      >
        <div
          style={
            Object {
              "color": "green",
            }
          }
        >
          1
           
          <span
            className="col-announcement-count"
          >
            Positive
          </span>
        </div>
        <div
          style={
            Object {
              "color": "red",
            }
          }
        >
          1
           
          <span
            className="col-announcement-count"
          >
            Negative
          </span>
        </div>
        <div
          className="color"
        >
          2
           
          <span
            className="assembl-icon-profil"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Announcement component <Announcement/> on a thread should render an announcement with the content associated to a thread idea 1`] = `
<div
  className="announcement"
>
  <div
    className="announcement-title"
  >
    <div
      className="title-hyphen"
    >
       
    </div>
    <h3
      className="announcement-title-text dark-title-1"
    >
      <span
        className={undefined}
        style={undefined}
      >
        Announcement
      </span>
    </h3>
  </div>
  <div
    className="announcement-media col-md-push-4 col-md-8 col-xs-12"
  >
    <section
      className="media-section background-light"
    >
      <div
        className="max-container"
      >
        <div
          className="container-fluid"
        >
          <div
            className="col-md-4 col-sm-12"
          >
            <div
              className="media-description"
            >
              <div
                className="media-description-icon"
              >
                <span
                  className="assembl-icon-pepite color2"
                >
                   
                </span>
              </div>
              <div
                className="description-txt"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": undefined,
                  }
                }
              />
              <div
                className="box-hyphen left"
              >
                 
              </div>
            </div>
          </div>
          <div
            className="col-md-8 col-sm-12"
          >
            <div
              className="media-right"
            >
              <div
                className="media-description-layer media-description-top"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "<p>Bonjour à tous</p>",
                  }
                }
              />
              <div
                className="media-description"
              >
                <div
                  className="media-description-icon"
                >
                  <span
                    className="assembl-icon-pepite color2"
                  >
                     
                  </span>
                </div>
                <div
                  className="description-txt"
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": undefined,
                    }
                  }
                />
                <div
                  className="box-hyphen left"
                >
                   
                </div>
              </div>
              <div
                className="media-description-layer media-description-bottom"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": null,
                  }
                }
              />
            </div>
          </div>
        </div>
      </div>
    </section>
  </div>
  <div
    className="col-md-pull-8 col-md-4 col-xs-12"
  >
    <div
      className="announcement-statistics"
    >
      <div
        className="announcement-doughnut"
      >
        <div
          className="statistics-doughnut"
        >
          <div
            className="statistics"
          >
            <div
              className="doughnut-container superpose"
            >
              <svg
                className="doughnut"
                viewBox="0 0 140 140"
              >
                <path
                  className="circle"
                  d="M 117.55282581475768 54.54915028125263 A 50 50 0 0 0 70 20"
                  fill="transparent"
                  onBlur={[Function]}
                  onClick={null}
                  onFocus={[Function]}
                  onMouseOut={[Function]}
                  onMouseOver={[Function]}
                  stroke="#46D081"
                />
                <path
                  className="circle"
                  d="M 113.30650522723651 94.99093045493377 A 50 50 0 0 0 117.55282581475768 54.54915028125263"
                  fill="transparent"
                  onBlur={[Function]}
                  onClick={null}
                  onFocus={[Function]}
                  onMouseOut={[Function]}
                  onMouseOver={[Function]}
                  stroke="#F2474D"
                />
                <path
                  className="circle"
                  d="M 49.65360166665506 24.32698745581962 A 50 50 0 1 0 113.30650522723651 94.99093045493377"
                  fill="transparent"
                  onBlur={[Function]}
                  onClick={null}
                  onFocus={[Function]}
                  onMouseOut={[Function]}
                  onMouseOver={[Function]}
                  stroke="#FAC16F"
                />
                <path
                  className="circle"
                  d="M 69.99999999999999 20 A 50 50 0 0 0 49.65360166665506 24.32698745581962"
                  fill="transparent"
                  onBlur={[Function]}
                  onClick={null}
                  onFocus={[Function]}
                  onMouseOut={[Function]}
                  onMouseOver={[Function]}
                  stroke="#9374FF"
                />
              </svg>
            </div>
            <div
              className="doughnut-label superpose"
            >
              <div
                className="doughnut-label-count"
              >
                15
              </div>
              
              <div
                className="doughnut-label-text"
              >
                <span
                  className={undefined}
                  style={undefined}
                >
                  Reactions
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="announcement-numbers"
      >
        <div
          className="counters announcement-numbers"
        >
          <span
            className="counter-with-icon"
          >
            2
             
            <span
              className="assembl-icon assembl-icon-message"
            />
          </span>
           - 
          <span
            className="counter-with-icon"
          >
            2
             
            <span
              className="assembl-icon assembl-icon-profil"
            />
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Announcement component createDoughnutElements function should return an array of elements for a <StatisticDoughnut /> component 1`] = `
Array [
  Object {
    "Tooltip": <Tooltip
      bsClass="tooltip"
      className="no-arrow-tooltip"
      id="dontUnderstandTooltip"
      placement="right"
>
      2
       
      <Translate
            value="debate.dontUnderstand"
      />
</Tooltip>,
    "color": "yellow",
    "count": 2,
  },
  Object {
    "Tooltip": <Tooltip
      bsClass="tooltip"
      className="no-arrow-tooltip"
      id="disagreeTooltip"
      placement="right"
>
      0
       
      <Translate
            value="debate.disagree"
      />
</Tooltip>,
    "color": "red",
    "count": 0,
  },
  Object {
    "Tooltip": <Tooltip
      bsClass="tooltip"
      className="no-arrow-tooltip"
      id="likeTooltip"
      placement="right"
>
      4
       
      <Translate
            value="debate.like"
      />
</Tooltip>,
    "color": "green",
    "count": 4,
  },
  Object {
    "Tooltip": <Tooltip
      bsClass="tooltip"
      className="no-arrow-tooltip"
      id="moreinfoTooltip"
      placement="right"
>
      3
       
      <Translate
            value="debate.moreinfo"
      />
</Tooltip>,
    "color": "purple",
    "count": 3,
  },
]
`;
